Sources/Core/GraphRequest.swift

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+//
+// You are hereby granted a non-exclusive, worldwide, royalty-free license to use,
+// copy, modify, and distribute this software in source code or binary form for use
+// in connection with the web services and APIs provided by Facebook.
+//
+// As with any software that integrates with the Facebook platform, your use of
+// this software is subject to the Facebook Developer Principles and Policies
+// [http://developers.facebook.com/policy/]. This copyright notice shall be
+// included in all copies or substantial portions of the software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import UIKit
+
+typealias GraphPath = String
+typealias GraphRequestDataAttachment = Data
+
+struct GraphRequest {
+
+  /// The HTTPMethod to use for a graph request
+  enum HTTPMethod: String {
+    case get = "GET"
+    case post = "POST"
+    case delete = "DELETE"
+  }
+
+  struct Flags: OptionSet {
+    let rawValue: Int
+
+    static let none: Flags = Flags(rawValue: 0)
+
+    /// indicates this request should not use a client token as its token parameter
+    static let skipClientToken: Flags = Flags(rawValue: 1 << 1)
+
+    /// indicates this request should not close the session if its response is an oauth error
+    static let doNotInvalidateTokenOnError: Flags = Flags(rawValue: 1 << 2)
+
+    /// indicates this request should not perform error recovery
+    static let disableErrorRecovery: Flags = Flags(rawValue: 1 << 3)
+  }
+
+  /// The Graph API endpoint to use for the request, for example "me".
+  let graphPath: GraphPath
+
+  /// The request parameters.
+  let parameters: [String: Any]
+
+  /**
+   The HTTPMethod to use for the request, access rawValues for encoding purposes
+   ex: HTTPMethod.get.rawValue is "GET".
+  */
+  let httpMethod: HTTPMethod
+
+  /// An optional access token used by the request.
+  let accessToken: AccessToken?
+
+  /**
+   The Graph API version to use (e.g., "v2.0")
+   Defaults to `Settings.graphAPIVersion` if not specified during initialization
+  */
+  let version: String
+
+  var flags: Flags
```

Comment:
```suggestion
  /// Insert docs?
  var flags: Flags
```

Reply:
I'm not sure exactly what to put there. There weren't any docs in FBSDKGraphRequest+Internal.swift where the enum is defined. Do you have any suggestions for what to say here? I'm hesitant to add a placeholder without a plan on how to resolve it...

Reply:
This actually may be a code smell. Just having a grab bag of flags instead of a more meaningful abstraction. We should discuss this more. Good catch!